MetadataExtractor/Formats/Jpeg/HuffmanTablesDirectory.cs

```diff
@@ -0,0 +1,323 @@
+        {
+            get
+            {
+                byte[] result = new byte[_lengthBytes.Length];
+                Array.Copy(_lengthBytes, 0, result, 0, _lengthBytes.Length);
+                return result;
+            }
+        }
+
+        /// <returns>A byte array with the V values for this table.</returns>
+        [NotNull]
+        public byte[] ValueBytes
+        {
+            get
+            {
+                byte[] result = new byte[_valueBytes.Length];
+                Array.Copy(_valueBytes, 0, result, 0, _valueBytes.Length);
+                return result;
+            }
+        }
+
+        /// <summary>Evaluates whether this table is a "typical" Huffman table.</summary>
+        /// <remarks>
+        /// "Typical" has a special meaning in this context as the JPEG standard
+        /// (ISO/IEC 10918 or ITU-T T.81) defines 4 Huffman tables that has been
+        /// developed from the average statistics of a large set of images with 8-bit
+        /// precision. Using these instead of calculating the optimal Huffman tables
+        /// for a given image is faster, and is preferred by many hardware encoders
+        /// and some hardware decoders.
+        /// <para/>
+        /// Even though the JPEG standard doesn't define these as "standard tables"
+        /// and requires a decoder to be able to read any valid Huffman tables, some
+        /// are in reality limited decoding images using these "typical" tables.
+        /// Standards like DCF(Design rule for Camera File system) and DLNA(Digital
+        /// Living Network Alliance) actually requires any compliant JPEG to use only
+        /// the "typical" Huffman tables.
+        /// <para/>
+        /// This is also related to the term "optimized" JPEG. An "optimized" JPEG is
+        /// a JPEG that doesn't use the "typical" Huffman tables.
+        /// </remarks>
+        /// <returns>Whether or not this table is one of the predefined "typical" Huffman tables.</returns>
+        public bool IsTypical()
+        {
+            if (TableClass == HuffmanTableClass.DC)
+            {
+                return
+                    Equals(_lengthBytes, HuffmanTablesDirectory.TypicalLuminanceDcLengths) &&
+                    Equals(_valueBytes, HuffmanTablesDirectory.TypicalLuminanceDcValues) ||
+                    Equals(_lengthBytes, HuffmanTablesDirectory.TypicalChrominanceDcLengths) &&
+                    Equals(_valueBytes, HuffmanTablesDirectory.TypicalChrominanceDcValues);
+            }
+            else if (TableClass == HuffmanTableClass.AC)
+            {
+                return
+                    Equals(_lengthBytes, HuffmanTablesDirectory.TypicalLuminanceAcLengths) &&
+                    Equals(_valueBytes, HuffmanTablesDirectory.TypicalLuminanceAcValues) ||
+                    Equals(_lengthBytes, HuffmanTablesDirectory.TypicalChrominanceAcLengths) &&
+                    Equals(_valueBytes, HuffmanTablesDirectory.TypicalChrominanceAcValues);
```

Comment:
I don't believe these eight `Equals` checks are actually checking the contents of the arrays within.

Reply:
This will be resolved with a new ByteArrayUtil.EqualTo method